File: app/src/main/java/com/example/weather/noteAdapter.java
package com.example.weather;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewParent;
import android.widget.BaseAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.List;

public class noteAdapter extends BaseAdapter {

    Context mactivity;
    List<noteClass> notes;

    public noteAdapter(Context mactivity, List<noteClass> notes) {
        this.mactivity = mactivity;
        this.notes = notes;
    }

    @Override
    public int getCount() {
        return notes.size();
    }

    @Override
    public Object getItem(int i) {
        if(notes.isEmpty()){
            return null;
        }else{
            return notes.get(i);
        }

    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View v;
        LayoutInflater inflater = (LayoutInflater) mactivity.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        v = inflater.inflate(R.layout.adapter_view_layout, null, false);

        TextView noteName = v.findViewById(R.id.textViewNoteName);
        TextView noteDate = v.findViewById(R.id.textViewNoteDate);

        noteClass note = (noteClass) this.getItem(i);

        if(notes.isEmpty()){
            noteName.setText("Empty");
        }else{
            noteName.setText(note.getNoteTitle());
        }

        if(notes.isEmpty()){
            noteDate.setText("Empty");
        }else{
            noteDate.setText(note.getNoteCreationDate());
        }
        return v;
    }
}

File: app/src/main/java/com/example/weather/Adapter/NoteAdapter.kt
package com.example.weather.Adapter

import android.content.Context
import com.example.weather.Classes.noteClass
import android.widget.BaseAdapter
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.weather.R
import android.widget.TextView

class noteAdapter(var mactivity: Context, var notes: List<noteClass>) : BaseAdapter() {

    override fun getCount(): Int {
        return notes.size
    }

    override fun getItem(i: Int): Any? {
        return if (notes.isEmpty()) {
            null
        } else {
            notes[i]
        }
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        val v: View
        val inflater = mactivity.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        v = inflater.inflate(R.layout.adapter_view_layout, null, false)

        val noteName = v.findViewById<TextView>(R.id.textViewNoteName)
        val noteDate = v.findViewById<TextView>(R.id.textViewNoteDate)
        val note = getItem(i) as noteClass

        if (notes.isEmpty()) {
            noteName.text = "Empty"
        } else {
            noteName.text = note.noteTitle
        }
        if (notes.isEmpty()) {
            noteDate.text = "Empty"
        } else {
            noteDate.text = note.noteCreationDate
        }
        return v
    }
}